tests/FakeItEasy.IntegrationTests/FakingClassesTests.cs

```diff
@@ -12,10 +11,13 @@ public class FakingClassesTests
         [Test]
         public void Should_be_able_to_get_a_fake_value_of_uri_type()
         {
-            using (Fake.CreateScope(new NullFakeObjectContainer()))
-            {
-                A.Fake<Uri>();
-            }
+            // Arrange
+
+            // Act
+            var fake = A.Fake<Uri>();
+
+            // Assert
+            Fake.GetFakeManager(fake).Should().NotBeNull("because we should be able to create a fake Uri");

```

Comment:
> The problem is that you can't (usually) catch a StackOverflowException... so it would be a bit pointless.

True, which, I suppose, is why the original test didn't attempt to catch and assert an exception, but just tried to create the fake and let any exception bubble up to the test runner.

I think I'd prefer that to the not-null assertion, which I don't think can *ever* fail.